packit/utils/obs_helper.py

```diff
@@ -0,0 +1,454 @@
+    meta configuration:
+
+    .. code-block:: xml
+
+       <repository name="images">
+         <path project="openSUSE:Factory" repository="containerfile"/>
+         <path project="openSUSE:Factory" repository="standard"/>
+         <arch>x86_64</arch>
+       </repository>
+
+    """
+
+    name: str
+    arch: list[str]
+    path: list[XmlPathEntry]
+
+
+def target_to_path(target: str) -> list[XmlPathEntry]:
+    """Converts a packit target name like ``fedora-rawhide-x86_64`` or
+    ``opensuse-leap-15.5`` to a list of xml path entries of the respective
+    projects on OBS. This function relies on the project setup on
+    build.opensuse.org and is not directly re-usable on other OBS instances.
+
+    """
+
+    target_split = target.split("-")
+    version, arch = target_split[-2:]
+    distro = "-".join(target_split[:-2])
+
+    if distro == "fedora":
+        if version == "rawhide":
+            return [XmlPathEntry(project="Fedora:Rawhide", repository="standard")]
+        return [
+            XmlPathEntry(project=f"Fedora:{version}", repository="standard"),
+            XmlPathEntry(project=f"Fedora:{version}", repository="update"),
+        ]
+
+    if distro == "epel":
+        if version == "9":
+            return [
+                XmlPathEntry(project=f"Fedora:EPEL:{version}", repository="stream"),
+            ]
+        if version in ("8", "7"):
+            return [
+                XmlPathEntry(project=f"Fedora:EPEL:{version}", repository="CentOS"),
+            ]
+
+    if distro == "opensuse-leap":
+        return [
+            XmlPathEntry(project=f"openSUSE:Leap:{version}", repository="standard"),
+        ]
+
+    if distro == "opensuse":
+        if arch == "x86_64":
+            return [XmlPathEntry(project="openSUSE:Factory", repository="snapshot")]
+        if arch in ("s390x", "aarch64", "ppc64le"):
+            postfix = {"s390x": "zSystem", "aarch64": "ARM", "ppc64le": "PowerPC"}[arch]
+            return [
+                XmlPathEntry(
+                    project=f"openSUSE:Factory:{postfix}",
+                    repository="standard",
+                ),
+            ]
+
+    raise ValueError(f"No preset available for {distro=}, {version=}, {arch=}")
+
+
+def targets_to_project_meta(
+    targets: list[str],
+    owner: str,
+    project_name: str,
+    description: Optional[str] = None,
+) -> ET.Element:
+    """Converts the list of packit targets (like `fedora-rawhide`) to a project
+    meta xml configuration for the respective project in OBS.
+
+    """
+    repos: list[OBSRepository] = []
+    for target in targets:
+        path = target_to_path(target)
+        arch = target.split("-")[-1]
+        name = target
+        added = False
+
+        for ind, repo in enumerate(repos):
+            if repo.path == path:
+                repos[ind] = OBSRepository(
+                    name=f"{repo.name}-{arch}",
+                    path=path,
+                    arch=[*repo.arch, arch],
```

Comment:
> However do we need to include the arches in the name?

It's kind of an obs thing. Take a look at the xml sent to obs
https://github.com/packit/packit/blob/3e1faa7c8083bf1106494b5ebbae4b63cd1086c7/tests/unit/test_obs_build.py#L60-L63

Reply:
> Listing all the arches feels a bit weird to me… And I have a feeling that it could also produce multiple project if user were to change the ordering of the arches in the list (since different order of arches ⇒ different name generated)…

Not at all. OBS has it's way of resolving arches for projects.

I have updated the smoke test to prove this https://github.com/packit/packit/blob/1cd4e87658b50ec2bcbe34e63f5f1a37b651c69d/.github/workflows/obs-build-smoke-test.yml#L40

Changing the order of the targets still resolves to the same project.

> Not to mention that this doesn't hold for Tumbleweed (Factory, as it has separate project for each arch…)

Same point here. OBS handles that